src/python/pants/backend/experimental/python/lockfile.py

```diff
@@ -272,21 +240,21 @@ async def generate_all_tool_lockfiles(
     # TODO(#12314): Add logic to inspect the Specs and generate for only relevant lockfiles. For
     #  now, we generate for all tools.
     requests = await MultiGet(
-        Get(PythonToolLockfileRequest, PythonToolLockfileSentinel, sentinel())
+        Get(PythonLockfileRequest, PythonToolLockfileSentinel, sentinel())
         for sentinel in union_membership.get(PythonToolLockfileSentinel)
     )
     if not requests:
         return ToolLockGoal(exit_code=0)
 
     results = await MultiGet(
-        Get(PythonToolLockfile, PythonToolLockfileRequest, req)
+        Get(PythonLockfile, PythonLockfileRequest, req)
         for req in requests
-        if req.lockfile_path not in {"<none>", "<default>"}
+        if req.dest not in {"<none>", "<default>"}
     )
     merged_digest = await Get(Digest, MergeDigests(res.digest for res in results))
     workspace.write_digest(merged_digest)
     for result in results:
-        logger.info(f"Wrote lockfile for {result.tool_name} to {result.path}")
+        logger.info(f"Wrote lockfile to {result.path}")
```

Comment:
This is a UX regression imo. I couldn't figure out a good, generic way to store this info on `PythonLockfile` because a `tool_name` property no longer makes sense.

Thoughts if this is worth trying to fix? It might be clear enough already what's happening.

Reply:
Okay, cool. I'll add a TODO in a followup to restore this once we have multiple user lockfiles.